ludwig/schema/parameter_metadata.py

```diff
@@ -0,0 +1,62 @@
+from dataclasses import dataclass
+from enum import Enum
+from typing import Any, List, Union
+
+
+class ExpectedImpact(Enum):
+    """The expected impact of determining a "good" value for a specific parameter.
+
+    - HIGH: this parameter should almost always be included in a hyperopt run and can make or break a good model.
+    - MEDIUM: this parameter can sometimes make or break a good model.
+    - LOW: this parameter usually does not have a significant impact on model performance.
+    """
+
+    UNKNOWN = 0
+    LOW = 1
+    MEDIUM = 2
+    HIGH = 3
+
+
+@dataclass
+class ParameterMetadata:
+    """Contains descriptive information that pertains to a Ludwig configuration parameter."""
+
+    # How this parameter can be displayed in a human-readable form.
+    ui_display_name: str = ""
+
+    # Why the default value for this parameter is the default.
+    default_value_reasoning: Union[str, None] = None
+
+    # Examples of other values that can be used for this parameter.
+    example_value: List[Any] = None
+
+    # List of related parameters that this parameter interacts with or depends on.
+    related_parameters: Union[List[str], None] = None
+
+    # Other information that is relevant for this parameter.
+    other_information: Union[str, None] = None
+
+    # If we change/increase/decrease this parameter, what's the intuition for how model performance would change, i.e.:
+    # learning curves, model speed, memory usage, etc.
+    description_implications: Union[str, None] = None
+
+    # What values would a machine learning expert suggest users try to help improve their model? Ideally, covers 95%
+    # (~2 sigma) of use cases.
+    suggested_values: Any = None
```

Comment:
Should this be a list?

Reply:
I'm inclined to leave this as Any for now, since it could be a list, set, range, or plain string (a verbal description)